source/modulo_new_core/src/translators/parameter_translators.cpp

```diff
@@ -106,120 +179,69 @@ std::shared_ptr<state_representation::ParameterInterface> read_parameter(const r
       // TODO: try clproto decode, re-use logic from above
       throw InvalidParameterException("Parameter byte arrays are not currently supported.");
     default:
-      // TODO: handle default
       break;
   }
   throw InvalidParameterException("Parameter " + parameter.get_name() + " could not be read!");
 }
 
-void read_parameter(
-    const rclcpp::Parameter& ros_parameter, std::shared_ptr<state_representation::ParameterInterface>& parameter
+std::shared_ptr<state_representation::ParameterInterface> read_parameter_const(
+    const rclcpp::Parameter& ros_parameter,
+    const std::shared_ptr<const state_representation::ParameterInterface>& parameter
```

Comment:
What part here would go wrong in this function if the second argument wasn't shared pointer of `const` ParameterInterface?

Reply:
Or would everything still work but we just want to be super explicit to be sure that the pointer is kept as the 'reference pointer' like you mentioned above?

Reply:
Nothing would go wrong, but by allowing calls to non-const methods it adds a risk that the value might be modified accidentally. This is just making everything more explicit.